VitalClient/src/main/java/io/tryvital/client/utils/LocalDateJsonAdapter.kt

```diff
@@ -0,0 +1,25 @@
+package io.tryvital.client.utils
+
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonReader
+import com.squareup.moshi.JsonWriter
+import java.time.LocalDate
+
+object LocalDateJsonAdapter: JsonAdapter<LocalDate>() {
+    override fun fromJson(reader: JsonReader): LocalDate? {
+        if (reader.peek() == JsonReader.Token.NULL) {
+            return reader.nextNull()
+        }
```

Comment:
What's this doing?

Reply:
Given a `YYYY-mm-dd` string, it decodes it into `java.time.LocalDate`.

This part is JSON null handling. [The implementation follows Moshi's inbuilt `Rfc3339DateJsonAdapter`](https://github.com/square/moshi/blob/master/moshi-adapters/src/main/java/com/squareup/moshi/adapters/Rfc3339DateJsonAdapter.kt).

Moshi is a JSON streaming parser so it doesn't work the same way as Swift Codable (which deserialises the entire JSON upfront).